tests/common/utils.py

```diff
@@ -66,3 +69,22 @@ def inverted_test(*arguments, **kwargs):
 
 
 fails = fails_with(AssertionError)
+
+
+def checks_deprecated_behaviour(func):
+    """A decorator for testing deprecated behaviour.
+
+    It will run the function once in non-strict mode (checking existing
+    test constraints), then once in strict mode (checking that a
+    deprecation exception is thrown).  This allows the pre-deprecation
+    tests to be retained with no change beyond the addition of a
+    decorator.
+
+    """
+    @functools.wraps(func)
+    def _inner(*args, **kwargs):
+        with settings(strict=True):
+            fails_with(HypothesisDeprecationWarning)(func)(*args, **kwargs)
+        with settings(strict=False):
+            func(*args, **kwargs)
```

Comment:
The block above is because I couldn't think of a good way to do that without warnings configuration I don't understand well enough to write correctly.